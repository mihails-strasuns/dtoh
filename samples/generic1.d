module sample;

void foo_ignored () { };

extern(C) {
    void foo ( ) { }

    int bar ( ) { return 0; }

    struct S
    {
        double d;
    }

    mixin("S foo_generated ( size_t x );");
}

alias Seq(T...) = T;

static foreach (int i, T; Seq!(S, int))
    mixin("extern(C) __gshared " ~ T.stringof ~ " global" ~ i.stringof ~ ";");


alias foo_t = extern(C) void function();

enum E
{
    A,
    B
}

extern(C):

float baz(E, foo_t);
