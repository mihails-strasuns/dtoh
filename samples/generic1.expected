#include <stdint.h>

// Used enum definitions:

enum E
{
    E_A = 0,
    E_B = 1
};

// Struct forward declaration to resolve cycles:

struct S;

// Function pointer types:

typedef void (*foo_t)();

// Used struct definitions:

struct S
{
    double d;
};

// Variable and function declarations:

void foo();
int32_t bar();
struct S foo_generated(uint32_t);
struct S global0;
int32_t global1;
float baz(enum E, foo_t);
